base_user_empty_password/models/res_users.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import fields, models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    has_password = fields.Boolean(compute="_compute_has_password")
+
+    def _compute_has_password(self):
+        # Bypass ORM as password is always empty in cache
+        self.env.cr.execute(
+            """SELECT id, password FROM res_users WHERE id IN %s;""", (tuple(self.ids),)
+        )
```

Comment:
for security reasons maybe we can execute query like this:
```
from psycopg2 import sql
query = sql.SQL("""
    SELECT {field} FROM res_partner WHERE ID IN %(partner_ids)s FOR UPDATE NOWAIT;
    UPDATE res_partner SET {field} = {field} + %(n)s
    WHERE id IN %(partner_ids)s
""").format(field=sql.Identifier(field))
self.env.cr.execute(query, {'partner_ids': tuple(self.ids), 'n': n})
```
it's just a copy paste from v14 source code